geopandas/io/parquet.py

```diff
@@ -0,0 +1,285 @@
+        a ``ValueError`` will be raised.
+    **kwargs
+        Any additional kwargs are passed to the engine.
+
+    Returns
+    -------
+    GeoDataFrame
+    """
+
+    import_optional_dependency(
+        "pyarrow", extra="pyarrow is required for parquet support."
+    )
+    from pyarrow import parquet
+
+    path, _, _, should_close = get_filepath_or_buffer(path)
+
+    kwargs["use_pandas_metadata"] = True
+    table = parquet.read_table(path, columns=columns, **kwargs)
+
+    df = table.to_pandas()
+
+    if should_close:
+        try:
+            path.close()
+        except:  # noqa: flake8
+            pass
+
+    metadata = None
+
+    try:
+        metadata = table.schema.metadata
+        if metadata is not None:
+            metadata = _decode_metadata(metadata.get(b"geo", b""))
+
+    except (TypeError, json.decoder.JSONDecodeError):
+        raise ValueError("Missing or malformed geo metadata in parquet file")
+
+    if metadata is None:
+        raise ValueError("Parquet file is missing required metadata 'geo' entry")
+
+    # Validate that required keys are present
+    required_keys = ("primary_column", "columns")
+    for key in required_keys:
+        if key not in metadata:
+            raise ValueError(
+                f"""'geo' metadata in parquet file is missing required key:
+                '{key}'"""
+            )
+
+    # Find all geometry columns that were read from the file.  May
+    # be a subset if 'columns' parameter is used.
+    geometry_columns = df.columns.intersection(metadata["columns"])
+
+    if not len(geometry_columns):
+        raise ValueError(
+            """No geometry columns are included in the columns read from
+            the parquet file.  To read this file without geometry columns,
+            use pandas.read_parquet() instead."""
+        )
+
+    column_metadata = metadata["columns"]
+
+    # Validate that geometry columns have required metadata and values
+    required_col_keys = ("crs", "encoding")
+    for col in geometry_columns:
+        for key in required_col_keys:
+            if key not in column_metadata[col]:
+                raise ValueError(
+                    f"""'geo' metadata in parquet file is missing required key
+                    {key} for column '{col}'"""
+                )
+
+        if column_metadata[col]["encoding"] != "WKB":
+            raise ValueError("Only WKB geometry encoding is supported")
+
+    geometry = metadata["primary_column"]
+
+    # Missing geometry likely indicates a subset of columns was read;
+    # promote the first available geometry to the primary geometry.
+    if len(geometry_columns) and geometry not in geometry_columns:
+        geometry = geometry_columns[0]
+
+    # Convert the WKB columns that are present back to geometry.
+    for col in geometry_columns:
+        df[col] = GeometryArray(
+            from_wkb(df[col].values), crs=column_metadata[col]["crs"]
```

Comment:
`from_wkb` already returns a GeometryArray, so you can leave out the wrapping in GeometryArray (and `from_wkb` now takes a crs keyword)